Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -97,7 +97,19 @@
      * The decay factor used for generating the weight of the point. For greater
      * values of lambda we become more biased in favor of recent points.
      */
-    private final double lambda;
+    private double lambda;
+    /**
+     * The last timestamp when lambda was changed
+     */
+    private long lastUpdateOflambda = 0;
+    /**
+     * most recent timestamp
+     */
+    private long mostRecentTimeStamp = 0;
```

Comment:
minor. the purpose of this variable is unclear. most recent time of sampling?

Reply:
The last change of lambda, which corresponds to maintaining the most recent time of sampling. Comments added.